web/docs/database/crud.md

```diff
@@ -1,3 +1,746 @@
+</Tabs>
+
+:::info Current Limitations
+In the default `create` and `update` implementations, we are saving all of the data that the client sends to the server. This is not always desirable, i.e. in the case when the client should not be able to modify all of the data in the entity.
+
+[In the future](#/docs/guides/crud#future-of-crud-operations-in-wasp), we are planning to add validation of action input, where only the data that the user is allowed to change will be saved.
+
+For now, the solution is to provide an override function. You can override the default implementation by using the `overrideFn` option and implementing the validation logic yourself.
+
+:::
+
+### Declaring a CRUD With All Available Options
+
+Here's an example of a more complex CRUD declaration:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```wasp title="main.wasp"
+crud Tasks { // crud name here is "Tasks"
+  entity: Task,
+  operations: {
+    getAll: {
+      isPublic: true, // optional, defaults to false
+    },
+    get: {},
+    create: {
+      overrideFn: import { createTask } from "@server/tasks.js", // optional
+    },
+    update: {},
+  },
+}
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```wasp title="main.wasp"
+crud Tasks { // crud name here is "Tasks"
+  entity: Task,
+  operations: {
+    getAll: {
+      isPublic: true, // optional, defaults to false
+    },
+    get: {},
+    create: {
+      overrideFn: import { createTask } from "@server/tasks.js", // optional
+    },
+    update: {},
+  },
+}
+```
+
+</TabItem>
+</Tabs>
+
+The CRUD declaration features the following fields:
+
+- `entity: Entity` <Required />
+
+  The entity to which the CRUD operations will be applied.
+
+- `operations: { [operationName]: CrudOperationOptions }` <Required />
+
+  The operations to be generated. The key is the name of the operation, and the value is the operation configuration.
+
+  - The possible values for `operationName` are:
+    - `getAll`
+    - `get`
+    - `create`
+    - `update`
+    - `delete`
+  - `CrudOperationOptions` can have the following fields:
+    - `isPublic: bool` - Whether the operation is public or not. If it is public, no auth is required to access it. If it is not public, it will be available only to authenticated users. Defaults to `false`.
+    - `overrideFn: ServerImport` - The import statement of the optional override implementation in Node.js.
+
+#### Defining the overrides
+
+Like with actions and queries, you can define the implementation in a Javascript/Typescript file. The overrides are functions that take the following arguments:
+
+- `args`
+
+  The arguments of the operation i.e. the data sent from the client.
+
+- `context`
+
+  Context contains the `user` making the request and the `entities` object with the entity that's being operated on.
+
+<ShowForTs>
+
+You can also import types for each of the functions you want to override from `@wasp/crud/{crud name}`. The available types are:
```

Comment:
I would wrap this Typescript info with `<ShowForTs>`

Reply:
It is, line 659 :smile: